### DISCUSSION TITLE
[question] Retrieving flag status for all records

### DISCUSSION THREAD

**Discussion by eleftrik** [spatie/laravel-model-flags#14] [Q&A]
First of all sorry for opening an issue, but Discussions is not enabled 😅

Let's pretend we have defined a `receivedMail` flag for our `User` model (not the best real world example, I know).

`User::find(1)->flag('receivedMail');`   // true
`User::find(2)->flag('receivedMail');`   // true
`User::find(3)->flag('receivedMail');`   // false

...and so on.

How can I retrieve (for example) all Users along with their flag status, without running n+1 queries?
Example use case: an administration panel that lists users and shows their `receivedMail` status.

Thank you

**Comment by BrekiTomasson** [comment]
Why not just run your query using `User::with('flags')` and then just iterate through them that way?
